wikipedia_transform/embedding_pipelines/open_ai_embedding_pipeline.py

```diff
@@ -0,0 +1,90 @@
+from collections.abc import Callable
+
+from langchain.docstore.document import Document
+from langchain.embeddings import CacheBackedEmbeddings
+from langchain.storage import LocalFileStore
+from langchain_community.vectorstores import FAISS
+from langchain_core.embeddings import Embeddings
+from langchain_openai import OpenAIEmbeddings
+from mypy_extensions import NamedArg
+
+from wikipedia_transform.embedding_pipelines import EmbeddingPipeline
+from wikipedia_transform.models import OpenAiEmbeddingModel, Record, settings
+from wikipedia_transform.models.types import OpenAiEmbeddingModelName, RecordType
+
+
+class OpenAiEmbeddingPipeline(EmbeddingPipeline):
+    """
+    A concrete implementation of EmbeddingPipeline that uses OpenAI's embedding models to convert
+    Records into embeddings.
+
+    Embeddings are stored at settings.output_path.
```

Comment:
Make this a parameter `output_path` rather than relying on settings fields.
The general principle here is that settings should be dispersed during construction.